find_email_addresses.py

```diff
@@ -0,0 +1,261 @@
+
+    Args:
+        domain: The string to test
+
+    Returns:
+        True if the domain is valid, false otherwise."""
+
+    # For some reason, urlparse won't recognize the domain without the scheme,
+    # so adding a scheme is a bit of a hack, but it works. It will even work
+    # if someone enters "http://domain.com" as the domain.
+    url = urllib.parse.urlparse("http://" + domain)
+    if url.path or url.params or url.query or url.fragment or not url.netloc:
+        return False
+
+    return True
+
+def get_emails_in_domain(domain, scheme='http', verbosity=0):
+    """Returns email addresses found in domain.
+
+    Iterates through publicly accessible pages/files found at
+    <scheme>://<domain> and collects all email addresses.
+
+    Args:
+        domain: The domain to search as a string
+        scheme: The scheme to use (default: http) as a string
+        verbosity: The level of verbosity (default: 0, max: 2)
+
+    Returns:
+        A list containing unique email addresses found
+
+    Raises:
+        DomainError: 'domain' is not a valid domain."""
+
+    if not _is_valid_domain(domain):
+        raise DomainError('"{}" is not a valid domain.'.format(domain))
+
+    emails = deque()
+    pages_visited = deque()
+    pages_to_visit = deque(['/'])
+
+    while len(pages_to_visit):
+        page_contents = ''
+
+        path = pages_to_visit.popleft()
+        link = _assemble_url(path, domain, scheme)
+        try:
+            if verbosity >= 2:
+                print('Processing page "{}"...'.format(link))
+            with urllib.request.urlopen(link) as page:
+                page_contents = page.read()
+        except urllib.error.HTTPError:
+            # Ignore errors from webpages
+            if verbosity >= 2:
+                print('Could not process page "{}".'.format(link))
+        pages_visited.append(path)
+
+        if page_contents:
+            page_contents = _decode_bytestring(page_contents)
+
+            for email in _get_emails_from_string(page_contents):
+                if not emails.count(email.lower()):
+                    emails.append(email.lower())
+
+            for link in _get_links_from_string(page_contents, domain):
+                if (not pages_to_visit.count(link)
+                        and not pages_visited.count(link)):
+                    pages_to_visit.append(link)
+
+    if verbosity >= 1:
+        print('Processed {} pages.'.format(len(pages_visited)))
+
+    return list(emails)
+
+def main():
+    """Handles user input from the command line."""
+
+    parser = ArgumentParser(
+        description='Find email addresses on pages from a given domain.')
+    parser.add_argument('--scheme', type=str, default='http',
+                        help='scheme to use (default: http)')
+    parser.add_argument('-v', '--verbose', default=0, action='count',
+                        help='increase verbosity')
+    parser.add_argument('domain', type=str, help='domain on which to search')
```

Comment:
The 'domain' argument does not appear to handle sub-domains properly.  Compare the following outputs:
```
python3 find_email_addresses.py  www.jana.com
Finding emails. This could take a while. Please wait...
No emails found at www.jana.com.
```
vs
```
python3 find_email_addresses.py  jana.com
Finding emails. This could take a while. Please wait...
support@wordpress.com
```